graylog2-web-interface/src/stores/metrics/MetricsStore.js

```diff
@@ -43,46 +42,41 @@ const MetricsStore = Reflux.createStore({
       return;
     }
     const metricsToFetch = {};
+
+    // First collect all node metric registrations
     Object.keys(this.registrations)
-      .filter((nodeId) => Object.keys(this.registrations[nodeId]).length > 0)
-      .forEach(nodeId => {
-        metricsToFetch[nodeId] = Object.keys(this.registrations[nodeId]).filter(metricName => this.registrations[nodeId][metricName] > 0);
-      });
-    const globalMetrics = Object.keys(this.globalRegistrations).filter(metricName => this.globalRegistrations[metricName] > 0);
-
-    if (this.nodes) {
-      Object.keys(this.nodes).forEach(nodeId => {
-        globalMetrics.forEach(metricName => {
-          if (!metricsToFetch[nodeId]) {
-            metricsToFetch[nodeId] = [];
-          }
-          if (metricsToFetch[nodeId].indexOf(metricName) == -1) {
-            metricsToFetch[nodeId].push(metricName);
-          }
-        });
+      .filter((nodeId) => Object.keys(this.registrations[nodeId].length > 0))
+      .forEach((nodeId) => {
+        Object.keys(this.registrations[nodeId])
+          .filter((metricName) => this.registrations[nodeId][metricName] > 0)
+          .forEach((metricName) => {
+            metricsToFetch[metricName] = 1;
+          });
       });
-    }
 
-    const promises = Object.keys(metricsToFetch)
-      .map((nodeId) => {
-        const url = URLUtils.qualifyUrl(ApiRoutes.ClusterMetricsApiController.multiple(nodeId).url);
-        const body = {metrics: metricsToFetch[nodeId]};
-        return fetch('POST', url, body).then((response) => {
-          const metrics = {};
-          response.metrics.forEach((metric) => metrics[metric.full_name] = metric);
-          return {nodeId: nodeId, metrics: metrics};
-        });
+    // Then collect all global metric registrations
+    Object.keys(this.globalRegistrations)
+      .filter((metricName) => this.globalRegistrations[metricName] > 0)
+      .forEach((metricName) => {
+        metricsToFetch[metricName] = 1;
       });
 
-    const promise = this._allResults(promises).then((responses) => {
+    const url = URLUtils.qualifyUrl(ApiRoutes.ClusterMetricsApiController.multipleAllNodes().url);
```

Comment:
But we still request the throughput counter for every node. So I think it's a trade off between fetching a bit more data from all nodes than needed and doing separate requests for each node.

For the throughput counter we contact every node anyway, so we won't even save any requests to any nodes. The only drawback I see is that we request a bit more data than needed.